Src/WitsmlExplorer.Api/Workers/CreateMudLogWorker.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Serilog;
+using Witsml;
+using Witsml.Data;
+using Witsml.Extensions;
+using Witsml.Query;
+using Witsml.ServiceReference;
+using WitsmlExplorer.Api.Jobs;
+using WitsmlExplorer.Api.Models;
+using WitsmlExplorer.Api.Services;
+
+namespace WitsmlExplorer.Api.Workers
+{
+    public interface ICreateMudLogWorker
+    {
+        Task<(WorkerResult, RefreshAction)> Execute(CreateMudLogJob job);
+    }
+
+    public class CreateMudLogWorker : ICreateMudLogWorker
+    {
+        private readonly IWitsmlClient witsmlClient;
+
+        public CreateMudLogWorker(IWitsmlClientProvider witsmlClientProvider)
+        {
+            witsmlClient = witsmlClientProvider.GetClient();
+        }
+
+        public async Task<(WorkerResult, RefreshAction)> Execute(CreateMudLogJob job)
+        {
+            var mudLog = job.MudLog;
+            Verify(mudLog);
+
+            var mudLogToCreate = SetupMudLogToCreate(mudLog);
+
+            var result = await witsmlClient.AddToStoreAsync(mudLogToCreate);
+            if (result.IsSuccessful)
+            {
+                await WaitUntilMudLogHasBeenCreated(mudLog);
+                Log.Information("{JobType} - Job successful", GetType().Name);
+                var workerResult = new WorkerResult(witsmlClient.GetServerHostname(), true, $"MudLog created ({mudLog.Name} [{mudLog.Uid}])");
+                var refreshAction = new RefreshWellbore(witsmlClient.GetServerHostname(), mudLog.WellUid, mudLog.Uid, RefreshType.Add);
+                return (workerResult, refreshAction);
+            }
+
+            var description = new EntityDescription { WellboreName = mudLog.WellboreName };
+            Log.Error($"Job failed. An error occurred when creating MudLog: {job.MudLog.PrintProperties()}");
+            return (new WorkerResult(witsmlClient.GetServerHostname(), false, "Failed to create MudLog", result.Reason, description), null);        }
+
+        private async Task WaitUntilMudLogHasBeenCreated(MudLog mudLog)
+        {
+            var isMudLogCreated = false;
+            var query = MudLogQueries.QueryById(mudLog.WellUid, mudLog.WellboreUid, mudLog.Uid);
+            var maxRetries = 30;
+            while (!isMudLogCreated)
+            {
+                if (--maxRetries == 0)
+                {
+                    throw new InvalidOperationException($"Not able to read newly created MudLog with name {mudLog.Name} (id={mudLog.Uid})");
+                }
+                Thread.Sleep(1000);
+                var mudLogResult = await witsmlClient.GetFromStoreAsync(query, OptionsIn.IdOnly);
+                isMudLogCreated = mudLogResult.MudLogs.Any();
+            }
+        }
+
+        private static WitsmlMudLogs SetupMudLogToCreate(MudLog mudLog)
+        {
+
```

Comment:
Nitpicking, but it is generally a lot of misplaced empty lines in this PR which should be cleaned up

Reply:
I think this is not just nitpicking. If code is merged without first being checked by Prettier or ESLint it will prevent others from commiting easily without either fixing it themselves or ignoring the automatic checks.